src/app/clusters/resource-monitoring-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,184 @@
+    {}
+
+    ResourceMonitoringCluster cluster;
+    ServerClusterRegistration registration;
+    ListEntry * next{ nullptr };
+};
+
+ListEntry * gActiveClusters{ nullptr };
+
+// Removes the first node with a matching pair (endpointId, clusterId) from the list and returns the new head of the list.
+ListEntry * removeListEntryFromList(ListEntry * head, EndpointId endpointId, ClusterId clusterId)
+{
+    ListEntry * current  = head;
+    ListEntry * previous = nullptr;
+
+    while (current)
+    {
+        if (current->cluster.GetEndpointId() == endpointId && current->cluster.GetClusterId() == clusterId)
+        {
+            CodegenDataModelProvider::Instance().Registry().Unregister(&(current->cluster));
+            ListEntry * temp = current;
+            if (previous)
+            {
+                previous->next = current->next;
+            }
+            else
+            {
+                head = current->next;
+            }
+            current = current->next;
+            delete temp;
+        }
+        else
+        {
+            previous = current;
+            current  = current->next;
+        }
+    }
+    return head;
+}
+
+} // namespace
+
+// Common helper for cluster initialization
+void InitResourceMonitoringCluster(EndpointId endpointId, ClusterId clusterId, BitFlags<ResourceMonitoring::Feature> featureFlags,
+                                   const ResourceMonitoring::ResourceMonitoringCluster::OptionalAttributeSet & optionalAttributeSet,
+                                   chip::app::Clusters::ResourceMonitoring::DegradationDirectionEnum degradationDirection,
+                                   bool resetConditionSupported)
+{
+    ListEntry * previous = nullptr;
+    ListEntry * current  = gActiveClusters;
+
+    while (current != nullptr)
+    {
+        previous = current;
+        current  = current->next;
+    }
+
+    ListEntry * newNode =
+        new ListEntry{ endpointId, clusterId, featureFlags, optionalAttributeSet, degradationDirection, resetConditionSupported };
+
+    CHIP_ERROR err = CodegenDataModelProvider::Instance().Registry().Register(newNode->registration);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "Failed to register cluster %u/" ChipLogFormatMEI ":   %" CHIP_ERROR_FORMAT, endpointId,
+                     ChipLogValueMEI(clusterId), err.Format());
+    }
+
+    if (previous)
+    {
+        previous->next = newNode;
+    }
+    else
+    {
+        gActiveClusters = newNode; // List was empty
+    }
+}
+
+//
+// HEPA Filter Monitoring Cluster
+//
+
+void MatterHepaFilterMonitoringClusterInitCallback(EndpointId endpointId)
+{
+    uint32_t optionalAttributeBits = 0;
+    ResourceMonitoring::ResourceMonitoringCluster::OptionalAttributeSet optionalAttributeSet(optionalAttributeBits);
+    uint32_t featureMap = CodegenClusterIntegration::LoadFeatureMap(endpointId, HepaFilterMonitoring::Id);
```

Comment:
FeatureMap is AAI-only (i.e. forced always callback) in https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/zcl.json#L584

So this means this does not work: there is no ember constant for this. It seems we have to preserve whatever the old API was having (I believe they set it in the constructor)

We have to figure out what needs to be read from ZAP vs what not (i.e. what is RAM vs callback).